Bluetooth: Tester: Send test data when entering streaming state

sjanc opened the pull request:
For now this seems to be required by PTS as there is not Upper Tester
action for this in Test Specification.

This is affecting few BAP/USR/STR qualifications tests

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/79059

sjanc committed:
Bluetooth: Tester: Send test data when entering streaming state

For now this seems to be required by PTS as there is not Upper Tester
action for this in Test Specification.

Signed-off-by: Szymon Janc <szymon.janc@codecoup.pl>

sjanc committed:
Bluetooth: Tester: Fix BTP response for BAP Send Data command

struct btp_bap_send_rp is already fixed size and contain extra
response byte.

Signed-off-by: Szymon Janc <szymon.janc@codecoup.pl>

sjanc commented:
Fix https://github.com/zephyrproject-rtos/zephyr/issues/79059

sjanc commented:
BAP/USR/STR/BV-365-C is stil failing, not sure yet why, other seem to pass

Thalley commented:
> BAP/USR/STR/BV-365-C is stil failing, not sure yet why, other seem to pass

Maybe it's unhappy that the data length is not equal to the SDU size?

fredrikdanebjer reviewed (dismissed):
Got a nice walkthrough from @MicImbriani.

Has BT-Sig been alerted on the tester not sending the test-cmd?

Im fine otherwise

Thalley commented on tests/bluetooth/tester/src/audio/btp_bap_unicast.c:
```diff
@@ -626,6 +626,7 @@ static void stream_started(struct bt_bap_stream *stream)
 {
 	struct btp_bap_unicast_stream *u_stream = stream_bap_to_unicast(stream);
 	struct bt_bap_ep_info info;
+	const uint8_t test_data[5] = { 0x01, 0x02, 0x03, 0x04, 0x05 };
```
Since we configure the SDU size as a product if Octets_Per_Codec_Frame, Audio_Channel_Allocation and Codec_Frame_Blocks_Per_SDU, I think the size of the test data we send should present that. 

Sending e.g. 5 octets when we've configured the SDU size to be e.g. 40 is invalid behavior, which we arguably should not do. (Ideally the data would also be LC3 encoded, but I don't know if PTS cares about that).

MicImbriani replied:
I can see your reasoning, however the test specification does not require this. In fact, in some tests (BAP/USR/STR/BV-369-C) it even specifies that Null PDUs can be valid: "The IUT transmits CIS Data PDUs or CIS Null PDUs"

Thalley replied:
What the test specs expect/demand may not always be what PTS expects/demands :D I think there was a test case that failed due to not sending the right amount of octets which is of course sort of a PTS bug, but in some cases it's just easier (or at least faster) to make PTS happy than to wait for a PTS fix, especially since BAP requires this (even if the test spec does not specifically mention it).

sjanc replied:
yeap, better to be compliant even if PTS is not validating it in all cases

sjanc committed:
Bluetooth: Tester: Set ISO MTU to 310

This makes sure we always fit all configurations. Worst case is two
audio streams in 48_6 configuration in single ISO channel (2x155).

Signed-off-by: Szymon Janc <szymon.janc@codecoup.pl>

sjanc commented:
All BAP/USR/STR are passing now

sjanc commented:
Fixed MESH DFU->DFUM and MBT->MBTM,  no other changes
Apparently export is not yet fixed in Qualification Workspace
